dotCMS/src/main/java/com/dotcms/util/HttpRequestDataUtil.java

```diff
@@ -209,4 +218,29 @@ public static String getHostname(final HttpServletRequest request) {
 		return hostName;
 	}
 
+	/**
+	 * Convenience method to find out the server port and cache it to use it must likely in logs.
+	 *
+	 * @return String representing the server port
+	 */
+    public static Optional<String> getServerPort() {
+    	if (serverPort == null) {
```

Comment:
Codacy found an issue: [Singleton is not thread safe](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6151243)